### PR TITLE
Implement tenant isolation in inference request management

### PR DESCRIPTION
**Pull request by kkaneda**

### CODE REVIEW

File: server/internal/infprocessor/infprocessor.go [hunk 1]
```diff
@@ -111,9 +115,9 @@ func (p *P) Run(ctx context.Context) error {
 }
 
 func (p *P) scheduleTask(ctx context.Context, t *Task) {
-	if len(p.enginesByID) == 0 {
+	engines := p.engines[t.TenantID]
+	if len(engines) == 0 {
 		t.ErrCh <- fmt.Errorf("no engine found")
-		return
```
Comment (Ladicle) [inline server/internal/infprocessor/infprocessor.go:116]: was this removal intentional?

**Review by Ladicle — APPROVED** [review]

File: server/internal/infprocessor/infprocessor.go (hunk 1 above)
Comment (kkaneda) [reply server/internal/infprocessor/infprocessor.go:116]: ah no.. let me add back.
